### ISSUE TITLE
Changing tileset on the main menu moans about layering ids not existing

### ISSUE THREAD

**Issue by Procyonae** [CleverRaven/Cataclysm-DDA#78592]
### Describe the bug

When you change tileset the newish layering stuff verifies ids except on the menu only /core has been loaded so it errors that the ids are invalid

### Attach save file

N/A

### Steps to reproduce

Change tileset/overmap tileset while on the menu with at least one ending as a tileset with custom layering specified (eg Ultice/MSX)
Get errors

### Expected behavior

No silly errors whether that's by not verifying the ids if /data/json isn't loaded or just not verifying them period bc we don't verify any other tileset sprite targets afaik (which would cause errors whenever we obsolete/migrate any id Edit: And wouldn't work period bc then you can't have modded stuff in tilesets unless you also specify the src in the tileset or something)

### Screenshots

Example error
```
 DEBUG    : Layering data: f_flagpole not a valid furniture/terrain object

 FUNCTION : void tileset_cache::loader::load_layers(const JsonObject&)
 FILE     : src/cata_tiles.cpp
 LINE     : 890
 VERSION  : 47f98df320-dirty
```

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.5247 (22H2)
- Game Version: 47f98df320-dirty [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth]
]

### Additional context

Notably we seem to reload or at least reverify the regular tileset when you change overmap tileset (incl when it wasn't shared) which seems incredibly inefficient

**Comment by RenechCDDA** [comment]
Oh yeah, got this earlier. Confirmed then.

**Comment by ShnitzelX2** [comment]
Yeah, that makes sense, I didn't account for changing tilesets in the title menu. I think that 
> not verifying the ids if /data/json isn't loaded

is probably the solution, so I'll try to at least patch this soon. 

Though, I don't think that switching tilesets in the title menu should functionally do anything but change what's loaded (instead of doing loading immediately after switching), so maybe I'll take a deeper look.

**Comment by Procyonae** [comment]
Outside of loading screens there's no real reason for tilesets to be loaded on the menu at all, making them only load what's necessary for the active mods would be pretty neat if that's doable ye
